server/consumer.go

```diff
@@ -1111,7 +1113,7 @@ func (o *Consumer) deliverCurrentMsg(subj string, msg []byte, seq uint64) bool {
 
 	// If we are partitioned and we do not match, do not consider this a failure.
 	// Go ahead and return true.
-	if o.config.FilterSubject != _EMPTY_ && subj != o.config.FilterSubject {
+	if o.config.FilterSubject != _EMPTY_ && !o.isFilteredMatch(subj) {
```

Comment:
I think this was missed in wildcard support perhaps?

Reply:
Seems so, great catch and thank you!